ext/uri/uri_parser_rfc3986.c

```diff
@@ -214,6 +279,26 @@ ZEND_ATTRIBUTE_NONNULL static zend_result php_uri_parser_rfc3986_port_read(const
 	return SUCCESS;
 }
 
+static zend_result php_uri_parser_rfc3986_port_write(struct uri_internal_t *internal_uri, zval *value, zval *errors)
+{
+	UriUriA *uriparser_uri = get_uri_for_writing(internal_uri);
+	int result;
+
+	if (Z_TYPE_P(value) == IS_NULL) {
+		result = uriSetPortTextMmA(uriparser_uri, NULL, NULL, mm);
+	} else {
+		ZVAL_STR(value, zend_long_to_str(Z_LVAL_P(value)));
+		result = uriSetPortTextMmA(uriparser_uri, Z_STRVAL_P(value), Z_STRVAL_P(value) + Z_STRLEN_P(value), mm);
```

Comment:
Storing the stringified long into the zval, just to pull it out of the zval sounds like a needless layer of indirection. You can just store it as a `zend_string *` directly. You are also modifying the value of the zval which can have unintended consequences.

This is also leaking memory. See this example:

```php
<?php

$uri = \Uri\Rfc3986\Uri::parse('https://example.com/');
$port = 123;
$newUri = $uri->withPort($port);
var_dump($newUri, $port);
```